Add 0.74 announcement post (#4043)
facebook/react-native-website: website/blog/2024-04-22-release-0.74.md

---
title: 'React Native 0.74 - Yoga 3.0, Bridgeless New Architecture, and more'
authors: [hurali, alanjhughes, alfonsocj, huntie, cortinico]
tags: [announcement, release, yoga]
date: 2024-04-22
---

Today we're releasing React Native 0.74! This release adds Yoga 3.0, Bridgeless by default under the New Architecture, batched `onLayout` updates (New Architecture), and Yarn 3 as the default package manager for new projects.

We are also removing deprecated APIs, with the removal of `PropTypes` and breaking changes to `PushNotificationIOS`. On Android, SDK 23 (Android 6.0) is now the minimum supported version.

### Highlights

- [Yoga 3.0](/blog/2024/04/22/release-0.74#yoga-30)
- [New Architecture: Bridgeless by Default](/blog/2024/04/22/release-0.74#new-architecture-bridgeless-by-default)
- [New Architecture: Batched `onLayout` Updates](/blog/2024/04/22/release-0.74#new-architecture-batched-onlayout-updates)
- [Yarn 3 for New Projects](/blog/2024/04/22/release-0.74#yarn-3-for-new-projects)

### Breaking Changes

> - [Android SDK minimum bump (Android 6.0)](/blog/2024/04/22/release-0.74#android-sdk-minimum-bump-android-60)

Reviewer: ```suggestion
- [Android minimum SDK bump (Android 6.0)](/blog/2024/04/22/release-0.74#android-sdk-minimum-bump-android-60)
```

<table>
<tr>
<td width="350">

```jsx
<View
  style={{
    backgroundColor: 'blue',
    width: 200,
    height: 200,
    flexDirection: 'row-reverse',
  }}>
  <View
    style={{
      backgroundColor: 'red',
      width: 100,
      height: 100,
      position: 'static',
    }}>
    <View
      style={{
        backgroundColor: 'green',
        width: 25,
        height: '25%',
        left: 25,
        top: 25,
        position: 'absolute',
      }}
    />
  </View>
</View>
```

</td>
<td width="300">

![Static Example](../static/blog/assets/0.74-static-example.png)

</td>
</tr>
</table>

Notice how the green `<View>` declares `left` and `top` and it is positioned relative to the blue `<View>`, not its parent.

React Native continues to default to `position: 'relative'` when no `position` is set.

### New Architecture: Bridgeless by Default

> In this release, we are making Bridgeless Mode the default when the New Architecture is enabled. You can learn more about our switch to Bridgeless as the default in this post (_coming soon_). To make the transition smoother we enhanced the interop layers to cover Bridgeless and worked with several libraries to make sure they will work in Bridgeless from day one.

Reviewer 2: ```suggestion
In this release, we are making Bridgeless Mode the default when the New Architecture is enabled. You can learn more about our switch to Bridgeless as the default in [this post](https://github.com/reactwg/react-native-new-architecture/discussions/174). To make the transition smoother we enhanced the interop layers to cover Bridgeless and worked with several libraries to make sure they will work in Bridgeless from day one.
```

> Bridgeless is not the only interop layer we worked on: we improved the New Renderer Interop layer too!

Reviewer 2: ```suggestion
Bridgeless is not the only interop layer we worked on: we improved the New Renderer Interop layers too! You can read more about them [here](https://github.com/reactwg/react-native-new-architecture/discussions/175)
```

:::danger
This change **may break code** that has relied on un-batched state updates. You'll need to refactor this code to use [updater functions](https://react.dev/learn/queueing-a-series-of-state-updates#updating-the-same-state-multiple-times-before-the-next-render) or equivalent.
:::

### Yarn 3 for New Projects

[Yarn 3](https://yarnpkg.com/blog/release/3.0) is now the default JavaScript package manager for new projects initialized with React Native Community CLI.

Yarn 3.x will be used with `nodeLinker: node-modules`, a mode providing compatibility with React Native libraries. This replaces Yarn Classic (1.x, deprecated) as the previous default. To upgrade Yarn version inside your existing app you can follow this [guide](https://yarnpkg.com/migration/guide).

```sh
$ yarn --help
━━━ Yarn Package Manager - 3.6.4 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  $ yarn <command>
```

The Community CLI also supports initializing projects with other package managers via the `--pm` flag ([read more](https://github.com/react-native-community/cli/blob/main/docs/init.md)).

## Breaking Changes

> ### Android SDK minimum bump (Android 6.0)

Reviewer: ```suggestion
### Android minimum SDK bump (Android 6.0)
```

## Breaking Changes

### Android SDK minimum bump (Android 6.0)

React Native 0.74 has a minimum Android SDK version requirement of 23 (Android 6.0). Previously, this was Android 5.0 (API 21). See our context for this change [here](https://github.com/react-native-community/discussions-and-proposals/discussions/740).

#### Bonus: Android app size reduction

The minimum SDK bump, together with several improvements at our native build, allowed us to greatly reduce the app size on user devices.

For example a newly created app with React Native 0.74 occupies ~13% less space on user device, resulting in ~4MB saved on device.

![Side-by-side comparison of a new React Native app in the Android system storage view](/blog/assets/0.74-android-app-size.jpg)

### Removal of deprecated `PropTypes`

Before 0.74, React Native was continuing to ship with `PropTypes`, an API that had been deprecated since React 15.5 in 2017! The presence of this deprecated code incurs costs in app size and memory overhead for apps that follow best practices.

> Starting in 0.74, React Native will no longer include the following `PropTypes` properties: `Image.propTypes`, `Text.propTypes`, `TextInput.propTypes`, `ColorPropType`, `EdgeInsetsPropType`, `PointPropType`, `ViewPropTypes` (see [commit](https://github.com/facebook/react-native/commit/228cb80af9ded20107f3c7a30ffe00e24471bfeb)).

Reviewer 3: @cortinico @yungsters Would it be fair to say (as a summary, preceding this) that 0.74 removes _**all built-in `PropTypes`**_ from React Native?

Reviewer 4: @huntie Yes, that is accurate.
